cmake/Modules/Platform/Emscripten.cmake

```diff
@@ -25,6 +25,11 @@ set_property(GLOBAL PROPERTY TARGET_SUPPORTS_SHARED_LIBS FALSE)
 # OpenCV) use this to detect bitness.
 set(CMAKE_SYSTEM_PROCESSOR x86)
 
+# Since emscripten produces Javascript, we set the byte order to little endian
+# to avoid cmake's checks.
```

Comment:
How about just "WebAssembly is always little endian so avoid the cmake's checks and force this."